video_xblock/mixins.py

```diff
@@ -165,6 +195,81 @@ def convert_3playmedia_caps_to_vtt(self, caps, video_id, lang="en", lang_label="
             response = {"lang": lang, "url": external_url, "label": lang_label}
         return response
 
+    def fetch_available_3pm_transcripts(self):
+        """
+        Fetch all available transcripts from 3PlayMedia API for current file ID.
+
+        :return: (generator of OrderedDicts) all transcript's data
+        """
+        results = self.get_available_3pm_transcripts(self.threeplaymedia_file_id, self.threeplaymedia_apikey)
+        errors = isinstance(results, dict) and results.get('errors')
+        log.debug("Fetched 3PM transcripts:\n{}".format(results))
+        if errors:
+            log.error("3PlayMedia transcripts fetching API request has failed!\n{}".format(
+                results.get('errors')
+            ))
+            raise StopIteration
+        for transcript_data in results:
+            transcript = self.fetch_3pm_translation(transcript_data)
+            transcript_ordered_dict = transcript._asdict()
+            transcript_ordered_dict['content'] = ''  # we don't want to parse it to JSON
+            yield transcript_ordered_dict
+
+    def get_available_3pm_transcripts(self, file_id, apikey):
+        """
+        Make API request to fetch list of available transcripts for given file ID.
+        """
+        domain = self.THREE_PLAY_MEDIA_API_DOMAIN
+        response = requests.get(
+            '{domain}files/{file_id}/transcripts?apikey={api_key}'.format(
+                domain=domain, file_id=file_id, api_key=apikey
+            )
+        )
+        if response.ok:
+            results = response.json()
+        else:
+            results = {"failed": True}
+        return results
+
+    def fetch_3pm_translation(self, transcript_data, format_id=TPMApiTranscriptFormatID.WEBVTT):
+        """
+        Fetch single transcript for given file ID in given format.
+
+        :param transcript_data:
+        :param format_id: defauts to VTT
+        :return: (namedtuple instance) transcript data
+        """
+        tid = transcript_data.get('id', '')
+        lang_id = transcript_data.get('language_id')
+        external_api_url = '{domain}files/{file_id}/transcripts/{tid}?apikey={api_key}&format_id={format_id}'.format(
+            domain=self.THREE_PLAY_MEDIA_API_DOMAIN,
+            file_id=self.threeplaymedia_file_id,
+            tid=tid,
+            api_key=self.threeplaymedia_apikey,
+            format_id=format_id
+        )
+        try:
+            content = requests.get(external_api_url).text
+        except Exception:  # pylint: disable=broad-except
+            log.exception(_("Transcript fetching failure: language [{}]").format(TPMApiLanguage(lang_id)))
+            return
```

Comment:
Please think about consistency in corner cases. Here you do return, but in def __init__(self, language_id): you do raise.